modules/openxr/extensions/openxr_fb_passthrough_extension_wrapper.cpp

```diff
@@ -0,0 +1,258 @@
+/* a copy of this software and associated documentation files (the       */
+/* "Software"), to deal in the Software without restriction, including   */
+/* without limitation the rights to use, copy, modify, merge, publish,   */
+/* distribute, sublicense, and/or sell copies of the Software, and to    */
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#include "openxr_fb_passthrough_extension_wrapper.h"
+
+#include "core/os/os.h"
+#include "scene/main/viewport.h"
+#include "scene/main/window.h"
+
+using namespace godot;
+
+OpenXRFbPassthroughExtensionWrapper *OpenXRFbPassthroughExtensionWrapper::singleton = nullptr;
+
+OpenXRFbPassthroughExtensionWrapper *OpenXRFbPassthroughExtensionWrapper::get_singleton() {
+	return singleton;
+}
+
+OpenXRFbPassthroughExtensionWrapper::OpenXRFbPassthroughExtensionWrapper(OpenXRAPI *p_openxr_api) :
+		OpenXRExtensionWrapper(p_openxr_api) {
+	request_extensions[XR_FB_PASSTHROUGH_EXTENSION_NAME] = &fb_passthrough_ext;
+	request_extensions[XR_FB_TRIANGLE_MESH_EXTENSION_NAME] = &fb_triangle_mesh_ext;
+	singleton = this;
+}
+
+OpenXRFbPassthroughExtensionWrapper::~OpenXRFbPassthroughExtensionWrapper() {
+	cleanup();
+}
+
+void OpenXRFbPassthroughExtensionWrapper::cleanup() {
+	fb_passthrough_ext = false;
+	fb_triangle_mesh_ext = false;
+}
+
+Viewport *OpenXRFbPassthroughExtensionWrapper::get_main_viewport() {
+	MainLoop *main_loop = OS::get_singleton()->get_main_loop();
+	if (!main_loop) {
+		print_error("Unable to retrieve main loop");
+		return nullptr;
+	}
+
+	auto *scene_tree = Object::cast_to<SceneTree>(main_loop);
+	if (!scene_tree) {
+		print_error("Unable to retrieve scene tree");
+		return nullptr;
+	}
+
+	Viewport *viewport = scene_tree->get_root()->get_viewport();
+	return viewport;
+}
+
+void OpenXRFbPassthroughExtensionWrapper::on_instance_created(const XrInstance instance) {
+	if (fb_passthrough_ext) {
+#ifdef DEBUG
+		Godot::print("OpenXR: Initializing passthrough extension...");
+#endif
```

Comment:
We don't use `DEBUG` in the Godot codebase. We do have `DEBUG_ENABLED` (for things that should run in the editor and in debug templates) and `DEV_ENABLED` (for dev builds only, not shipped in production).

Also not sure what this `Godot::print` is? We don't have this anywhere.

See `core/string/print_string.h` for print methods.

Reply:
@akien-mga @BastiaanOlij These were accidentally left in the code from the Godot 3.x implementation of `XRFbPassthroughExtensionWrapper`. I think that for now we do not need these initialization logs. I removed them so that we only log if some error happened.

Reply:
Yup, those are leftover from earlier days. If we do want to keep some I believe the correct method to use is print_verbose which will automatically silence unless you specifically ask for the output.